### ISSUE TITLE
Memory requirement in TF

### ISSUE THREAD

**Issue by olegarch** [bermanmaxim/LovaszSoftmax#22]
I was using batch of size 32 and once I switched to lovasz loss I had to decrease batch size to 8 to be able to train the same model. Otherwise OOM error is thrown.
Is it expected?

**Comment by bermanmaxim** [comment]
Hello,
Normally the memory usage should be around the same, but I only experimented with PyTorch. Maybe the sort algorithm is taking more memory than necessary. I might investigate some time in the future if I have time.

**Comment by marcocaccin** [comment]
Can confirm that Lovasz loss is eating up a lot more memory than, say, cross-entropy. This is on TF 2.1. I'll try to see if there's any low-hanging fruits for this contributions-welcome tag ;-)
